pkg/git/libgit2/managed/const.go

```diff
@@ -24,4 +24,65 @@ const (
 	// PathMaxLength represents the max length for the path element
 	// when cloning Git repositories via SSH.
 	PathMaxLength = 4096
+
+	// HTTPProtocol represents the HTTP protocol.
+	HTTPProtocol string = "http"
+
+	// HTTPSProtocol represents the HTTPS protocol.
+	HTTPSProtocol string = "https"
+
+	// SSHProtocol represents the SSH protocol.
+	SSHProtocol string = "ssh"
+
+	// SSHGitProtocol represents the ssh+git protocol.
+	SSHGitProtocol string = "ssh+git"
+
+	// GitSSHProtocol represents the git+ssh protocol.
+	GitSSHProtocol string = "git+ssh"
+
+	// HTTPManagedProtocol represents the HTTP managed protocol.
+	// It can be used to opt-in HTTP URLs over Managed Transport.
+	HTTPManagedProtocol string = "http+managed"
+
+	// HTTPUnmanagedProtocol represents the HTTP unmanaged protocol.
+	// It can be used to opt-out HTTP URLs over Managed Transport.
+	HTTPUnmanagedProtocol string = "http+unmanaged"
+
+	// HTTPSManagedProtocol represents the HTTPS managed protocol.
+	// It can be used to opt-in HTTPS URLs over Managed Transport.
+	HTTPSManagedProtocol string = "https+managed"
+
+	// HTTPSUnmanagedProtocol represents the HTTPS unmanaged protocol.
+	// It can be used to opt-out HTTPS URLs over Managed Transport.
+	HTTPSUnmanagedProtocol string = "https+unmanaged"
+
+	// SSHManagedProtocol represents the SSH managed protocol.
+	// It can be used to opt-in SSH URLs over Managed Transport.
+	SSHManagedProtocol string = "ssh+managed"
+
+	// SSHUnmanagedProtocol represents the SSH unmanaged protocol.
+	// It can be used to opt-out SSH URLs over Managed Transport.
+	SSHUnmanagedProtocol string = "ssh+unmanaged"
+)
+
+var (
+	// denySSHAutoUpgradeDomains is a list of domains that cannot be
+	// supported in managed transport via SSH.
+	denySSHAutoUpgradeDomains = []string{
+		// DevOps requires the Git protocol capabilities (e.g. multi_ack
+		// and multi_ack_detailed) that are not fully supported by libgit2/git2go
+		// in managed transport mode.
+		"dev.azure.com",
```

Comment:
Does this means we’ll not be able to get rid of libssh and openssl dependencies due to Azure DevOps?

Reply:
For the time being, yes. I will create an issue to further detail the problem and potential consequences.